File: SDKs/twilio.com/twilio_bulkexports_v1/1.36.0/python/src/sdk/models/operations/listexportcustomjob.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk.models import shared
LIST_EXPORT_CUSTOM_JOB_SERVERS = [
	"https://bulkexports.twilio.com",
]


@dataclass
class ListExportCustomJobPathParams:
    resource_type: str = field(default=None, metadata={'path_param': { 'field_name': 'ResourceType', 'style': 'simple', 'explode': False }})
    

@dataclass
class ListExportCustomJobQueryParams:
    page_size: Optional[int] = field(default=None, metadata={'query_param': { 'field_name': 'PageSize', 'style': 'form', 'explode': True }})
    

@dataclass
class ListExportCustomJobSecurity:
    account_sid_auth_token: shared.SchemeAccountSidAuthToken = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class ListExportCustomJobRequest:
    server_url: Optional[str] = field(default=None)
    path_params: ListExportCustomJobPathParams = field(default=None)
    query_params: ListExportCustomJobQueryParams = field(default=None)
    security: ListExportCustomJobSecurity = field(default=None)
    

@dataclass_json
@dataclass
class ListExportCustomJob200ApplicationJSONMeta:
    first_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'first_page_url' }})
    key: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'key' }})
    next_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'next_page_url' }})
    page: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'page' }})
    page_size: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'page_size' }})
    previous_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'previous_page_url' }})
    url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'url' }})
    

@dataclass_json
@dataclass
class ListExportCustomJob200ApplicationJSONListExportCustomJobResponse:
    jobs: Optional[List[shared.BulkexportsV1ExportExportCustomJob]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'jobs' }})
    meta: Optional[ListExportCustomJob200ApplicationJSONMeta] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'meta' }})
    

@dataclass
class ListExportCustomJobResponse:
    content_type: str = field(default=None)
    list_export_custom_job_response: Optional[ListExportCustomJob200ApplicationJSONListExportCustomJobResponse] = field(default=None)
    status_code: int = field(default=None)
    

File: SDKs/twilio.com/twilio_bulkexports_v1/1.36.0/typescript/src/sdk/models/operations/listexportcustomjob.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const LISTEXPORTCUSTOMJOB_SERVERS = [
	"https://bulkexports.twilio.com",
];



export class ListExportCustomJobPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=ResourceType" })
  resourceType: string;
}


export class ListExportCustomJobQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=PageSize" })
  pageSize?: number;
}


export class ListExportCustomJobSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class ListExportCustomJobRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  pathParams: ListExportCustomJobPathParams;

  @Metadata()
  queryParams: ListExportCustomJobQueryParams;

  @Metadata()
  security: ListExportCustomJobSecurity;
}


export class ListExportCustomJobListExportCustomJobResponseMeta extends SpeakeasyBase {
  @Metadata({ data: "json, name=first_page_url" })
  firstPageUrl?: string;

  @Metadata({ data: "json, name=key" })
  key?: string;

  @Metadata({ data: "json, name=next_page_url" })
  nextPageUrl?: string;

  @Metadata({ data: "json, name=page" })
  page?: number;

  @Metadata({ data: "json, name=page_size" })
  pageSize?: number;

  @Metadata({ data: "json, name=previous_page_url" })
  previousPageUrl?: string;

  @Metadata({ data: "json, name=url" })
  url?: string;
}


export class ListExportCustomJobListExportCustomJobResponse extends SpeakeasyBase {
  @Metadata({ data: "json, name=jobs", elemType: shared.BulkexportsV1ExportExportCustomJob })
  jobs?: shared.BulkexportsV1ExportExportCustomJob[];

  @Metadata({ data: "json, name=meta" })
  meta?: ListExportCustomJobListExportCustomJobResponseMeta;
}


export class ListExportCustomJobResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  listExportCustomJobResponse?: ListExportCustomJobListExportCustomJobResponse;

  @Metadata()
  statusCode: number;
}
